core/src/main/scala/com/nvidia/spark/rapids/tool/analysis/AppSparkMetricsAnalyzer.scala

```diff
@@ -320,6 +321,89 @@ class AppSparkMetricsAnalyzer(app: AppBase) extends AppAnalysisBase(app) {
     sqlRows.toSeq
   }
 
+  /**
+   * Aggregates the diagnostic SparkMetrics by stage.
+   * @param index the App-index (used by the profiler tool)
+   * @return sequence of StageDiagnosticAggTaskMetricsProfileResult
+   */
+  def aggregateDiagnosticSparkMetricsByStage(index: Int,
+      analyzerInput: Option[AppSQLPlanAnalyzer] = None):
+        Seq[StageDiagnosticMetricsProfileResult] = {
+    def bytesToMB(numBytes: Long): Long = numBytes / (1024 * 1024)
+    val sqlAnalyzer = analyzerInput match {
+      case Some(res) => res
+      case None => app.asInstanceOf[ApplicationInfo].planMetricProcessor
+    }
+    // TODO: this has stage attempts. we should handle different attempts
+    app.stageManager.getAllStages.map { sm =>
```

Comment:
Can we add a comment when this is expected to be none?

Reply:
updated, thanks!